plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1581,7 +1581,12 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         Snapshot snapshot = requireNonNull(icebergTable.snapshot(optimizeHandle.snapshotId().get()), "snapshot is null");
         rewriteFiles.validateFromSnapshot(snapshot.snapshotId());
         commit(rewriteFiles, session);
-        transaction.commitTransaction();
+        try {
+            transaction.commitTransaction();
```

Comment:
There are a few other places where `transaction.commitTransaction();` in `IcebergMetadata` is being called. Do they need wrapping as well?